homeassistant/components/insteon/api/aldb.py

```diff
@@ -0,0 +1,309 @@
+"""Web socket API for Insteon devices."""
+
+from pyinsteon import devices
+from pyinsteon.constants import ALDBStatus
+from pyinsteon.topics import (
+    ALDB_STATUS_CHANGED,
+    DEVICE_LINK_CONTROLLER_CREATED,
+    DEVICE_LINK_RESPONDER_CREATED,
+)
+from pyinsteon.utils import subscribe_topic, unsubscribe_topic
+import voluptuous as vol
+
+from homeassistant.components import websocket_api
+from homeassistant.core import HomeAssistant, callback
+
+from ..const import DEVICE_ADDRESS, ID, INSTEON_DEVICE_NOT_FOUND, TYPE
+from .device import async_device_name, notify_device_not_found
+
+ALDB_RECORD = "record"
+ALDB_RECORD_SCHEMA = vol.Schema(
+    {
+        vol.Required("mem_addr"): int,
+        vol.Required("in_use"): bool,
+        vol.Required("group"): vol.Range(0, 255),
+        vol.Required("is_controller"): bool,
+        vol.Optional("highwater"): bool,
+        vol.Required("target"): str,
+        vol.Optional("target_name"): str,
+        vol.Required("data1"): vol.Range(0, 255),
+        vol.Required("data2"): vol.Range(0, 255),
+        vol.Required("data3"): vol.Range(0, 255),
+        vol.Optional("dirty"): bool,
+    }
+)
+
+
+async def async_aldb_record_to_dict(dev_registry, record, dirty=False):
+    """Convert an ALDB record to a dict."""
+    return ALDB_RECORD_SCHEMA(
+        {
+            "mem_addr": record.mem_addr,
+            "in_use": record.is_in_use,
+            "is_controller": record.is_controller,
+            "highwater": record.is_high_water_mark,
+            "group": record.group,
+            "target": str(record.target),
+            "target_name": await async_device_name(dev_registry, record.target),
+            "data1": record.data1,
+            "data2": record.data2,
+            "data3": record.data3,
+            "dirty": dirty,
+        }
+    )
+
+
+async def async_reload_and_save_aldb(hass, device):
+    """Add default links to an Insteon device."""
+    if device == devices.modem:
+        await device.aldb.async_load()
+    else:
+        await device.aldb.async_load(refresh=True)
+    await devices.async_save(workdir=hass.config.config_dir)
+
+
+@websocket_api.websocket_command(
+    {vol.Required(TYPE): "insteon/aldb/get", vol.Required(DEVICE_ADDRESS): str}
+)
+@websocket_api.require_admin
+@websocket_api.async_response
+async def websocket_get_aldb(
+    hass: HomeAssistant,
+    connection: websocket_api.connection.ActiveConnection,
+    msg: dict,
+) -> None:
+    """Get the All-Link Database for an Insteon device."""
+    device = devices[msg[DEVICE_ADDRESS]]
+    if not device:
+        notify_device_not_found(connection, msg, INSTEON_DEVICE_NOT_FOUND)
+        return
+
+    # Convert the ALDB to a dict merge in pending changes
+    aldb = {mem_addr: device.aldb[mem_addr] for mem_addr in device.aldb}
+    aldb.update(device.aldb.pending_changes)
+    changed_records = list(device.aldb.pending_changes.keys())
+
+    dev_registry = await hass.helpers.device_registry.async_get_registry()
```

Comment:
Please import and use `async_get` from the device registry helper to get the device registry instead. The currently used helper is legacy and we prefer importing names over accessing `hass.helpers` to help typing.